src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncStrategy.java

```diff
@@ -0,0 +1,53 @@
+package com.amazonaws.services.kinesis.clientlibrary.lib.worker;
+
+/**
+ * Facade of methods that can be invoked at different points
+ * in KCL application execution to perform certain actions related to shard-sync.
+ */
+public interface ShardSyncStrategy {
+
+    enum StrategyType {
+        PERIODIC,
+        SHARD_END
```

Comment:
How is initial ShardSync (after deployment etc..) controlled with this change? It's running shardSync on all hosts independent of the StrategyType according to the ShardSyncAtStartup configuration?

Reply:
We were discussing this yesterday - since all workers would ShardSync at initialization, it leads to throttling from DescribeStream for large table streams. This further delays initialization. One option that came up was to do it probabilistically so only a fraction of workers do it initially.
We can take it up as a future optimization. Let me know if you have any other ideas.